libimage/pull.go

```diff
@@ -71,6 +72,38 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 	if err != nil {
 		return nil, err
 	}
+	rwDirs := func() []string {
+		rw := []string{}
+		for _, d := range []string{
+			r.store.RunRoot(),
+			r.store.GraphRoot(),
+			"/tmp",
+			"/var/tmp",
+			defaultConfig.Engine.ImageCopyTmpDir,
+			defaultConfig.Engine.TmpDir,
+		} {
+			if _, err := os.Stat(d); err == nil {
+				rw = append(rw, d)
+			}
+		}
+		return rw
+	}
+
+	if err := landlock.V5.BestEffort().RestrictPaths(
+		landlock.RODirs("/"),
+		landlock.RWDirs(rwDirs()...),
+	); err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := landlock.V5.BestEffort().RestrictPaths(
+			landlock.RWDirs("/"),
+		)
+		if err != nil {
+			logrus.Errorf("failed to reset landlock to read/write: %v", err)
```

Comment:
Yes, you can't really reset it, it is by design. That would be a way to break out, after all.

Reply:
Yup, I was looking at this differently. More like I want this thread to be locked down while it does the Pull and then able to unlock it when the thread finishes. Understand that the code could somehow use other parts of the shared memory to influence the parent thread, but it still would create a bigger barrier. The way landlock is being done is to prevent a hacked process from breaking out.  In this case we are just trying to prevent potential bugs in software being attacked via symbolic link attacks from file content being copied, untarred.

SELinux has the same concept but still allows you to change the type of a forked process, but it can control whether the process can change its label back.  It is not as good security wise as a full fork/exec, but it is still better then nothing.